ddd-koans/Value Object Pt. 2 & Entity/Collection Aversion/src/Main.kt

```diff
@@ -0,0 +1,50 @@
+import NoteValue.*
+import java.math.BigDecimal
+import java.math.MathContext
+import java.time.Duration
+
+data class Notes(private val notes: List<Note>) : LengthInTempo {
+    override fun lengthIn(tempo: Tempo): Length {
+        return notes
+            .map { it.lengthIn(tempo) }
+            .fold(Length(0.toBigDecimal())) { sum, length -> sum + length }
+    }
+}
+
+data class Note(val noteValue: NoteValue, val pitch: Pitch) : LengthInTempo by noteValue
+
+enum class NoteValue(private val relativeValue: Int): LengthInTempo {
+    WholeNote(1),
+    HalfNote(2),
+    QuarterNote(4),
+    EighthNote(8),
+    SixteenthNote(16),
+    ThirtySecondNote(32);
+
+    private val lengthInBeats: BigDecimal = 4.toBigDecimal().divide(relativeValue.toBigDecimal())
+
+    override fun lengthIn(tempo: Tempo): Length {
+        val minuteAsSeconds = Duration.ofMinutes(1L).toSeconds().toBigDecimal()
+        val durationInSeconds = minuteAsSeconds
```

Comment:
weird, on my environment `toSeconds` can't be used as it's private in `Duration`, instead we can use `seconds` property :man_shrugging: